src/outcomeFactory.js

```diff
@@ -0,0 +1,41 @@
+const FinalOutcome = require('./finalOutcome')
+const ActionableOutcome = require('./actionableOutcome')
+const NullOutcome = require('./nullOutcome')
+const MarkerFinder = require('./markerFinder')
+
+module.exports = class OutcomeFactory {
```

Comment:
I added this Factory (and I think this was being hinted at during the IPM) so that instead of returning a single, primitive value after going through the board - I return an outcome with information that other components can act on.

However now this factory is used by both my GameScenario and must be used on the top level (right now in setup.js). Is this a design flaw?